src/array.cpp

```diff
@@ -25,14 +25,14 @@ Array::Array(InBuffer &frame)
         charsToRead -= 1;
 
         // read the field type and construct the field
-        Field *field = Field::decode(frame);
+        auto field = Field::decode(frame);
         if (!field) continue;
 
         // less bytes to read
         charsToRead -= (uint32_t)field->size();
 
         // add the additional field
-        _fields.push_back(std::shared_ptr<Field>(field));
+        _fields.push_back(move(field));
```

Comment:
should this not be std::move()?

Reply:
Not necessarily due to ADL :) depends on what the code-style wants I suppose. I've added the std:: prefix.